### PR TITLE
add font modifier cell options: bold, italic, underline.

### PR DESCRIPTION
**Pull request by dyc3**
This PR adds cell options for the font modifiers﻿ bold, italic, and underline.

`tcell` was upgraded 1.4.0.

Strikethrough cell option was added as a no-op until tcell is updated to v2.0.0. I figured it wasn't a good idea to upgrade tcell by a whole major version without testing.

Closes #251

### CODE REVIEW

File: terminal/termbox/cell_options.go [hunk 1]
```diff
@@ -28,7 +28,16 @@ func cellColor(c cell.Color) tbx.Attribute {
 
 // cellOptsToFg converts the cell options to the termbox foreground attribute.
 func cellOptsToFg(opts *cell.Options) tbx.Attribute {
-	return cellColor(opts.FgColor)
+	a := cellColor(opts.FgColor)
+	if opts.Bold {
+		a |= tbx.AttrBold
+	}
+	// FIXME: Termbox doesn't have an italics attribute
```
Comment (mum4k) [inline terminal/termbox/cell_options.go:35]: How would you feel about returning an error here when the user tried to set italics or strikethrough on termbox? Giving them complete information in an error message might be more user friendly than silently ignoring the requested options.

File: cell/cell.go [hunk 2]
```diff
@@ -23,8 +23,12 @@ type Option interface {
 
 // Options stores the provided options.
 type Options struct {
-	FgColor Color
-	BgColor Color
+	FgColor       Color
```
Comment (mum4k) [inline cell/cell.go:26]: Would you be also willing to add test coverage for the modified modules? There are pre-existing unit test that should provide some guidance.

**Review by mum4k — CHANGES_REQUESTED** [review]
Looks great, thank you for the contribution!

Can you please sign the CLA? In addition to that, can you please respond indicating that you are giving me permission to merge this into the master branch?

**Comment by dyc3** [comment]
I thought I did sign the cla... I'll try it again

**Comment by dyc3** [comment]
@googlebot I signed it!

File: cell/cell.go (hunk 2 above)
Comment (dyc3) [reply cell/cell.go:26]: Sure thing!

File: terminal/termbox/cell_options.go (hunk 1 above)
Comment (dyc3) [reply terminal/termbox/cell_options.go:35]: Yeah, that sounds a lot better.

**Comment by dyc3** [comment]
> can you please respond indicating that you are giving me permission to merge this into the master branch?

I give you permission to merge this into the master branch.

**Comment by mum4k** [comment]
Thank you for the permission. We will merge this into the devel branch first and it will go into the next release together with a few minor changes.

**Comment by mum4k** [comment]
@dyc3 please let me know if you are still working on this, or if it is ready for merging. There is still some missing coverage on some of the added code, but I will be happy to add those tests as a follow-up if you prefer that.

**Comment by dyc3** [comment]
I'm currently done working on this. I looked at the missing coverage, and I thought the behavior that wasn't being covered was pretty trivial and wouldn't benefit too much from tests.

Although looking at it again, should the error paths be covered in tests?

**Comment by mum4k** [comment]
I would cover the error paths that we can cover easily. But there certainly are some not worth the work, because they depend on the termbox object being instantiated and we don't have a pre-established way of doing that so far.

I think this is good, I propose to merge it as it is and I can try to iterate on the test coverage a bit more before the merge to master.

Thank you very much for improving Termdash. I will aim towards upgrading tcell to 2.x next and then we can add support for the additional font modifiers. Please feel free to let me know if you would like to help with either of the two.

**Comment by dyc3** [comment]
Agree, lets merge it as-is. 

I'd like to help out with both. Feel free to @ me for whatever needs to get done.

**Comment by mum4k** [comment]
Sounds good and will do.

**Review by mum4k — APPROVED** [review]

**Pull request merged**